fe/fe-core/src/main/java/org/apache/doris/datasource/property/storage/HDFSProperties.java

```diff
@@ -0,0 +1,195 @@
+            description = "The xml files of Hadoop configuration.")
+    private String hadoopConfigResources = "";
+
+    @ConnectorProperty(names = {"hdfs.impersonation.enabled"},
+            required = false,
+            supported = false,
+            description = "Whether to enable the impersonation of HDFS.")
+    private boolean hdfsImpersonationEnabled = false;
+
+    @ConnectorProperty(names = {"fs.defaultFS"}, required = false, description = "")
+    private String fsDefaultFS = "";
+
+    /**
+     * The final HDFS configuration map that determines the effective settings.
+     * Priority rules:
+     * 1. If a key exists in `overrideConfig` (user-provided settings), its value takes precedence.
+     * 2. If a key is not present in `overrideConfig`, the value from `hdfs-site.xml` or `core-site.xml` is used.
+     * 3. This map should be used to read the resolved HDFS configuration, ensuring the correct precedence is applied.
+     */
+    Map<String, String> finalHdfsConfig;
+
+    public HDFSProperties(Map<String, String> origProps) {
+        super(Type.HDFS, origProps);
+        loadFinalHdfsConfig(origProps);
+    }
+
+    public static boolean guessIsMe(Map<String, String> props) {
+        if (MapUtils.isEmpty(props)) {
+            return false;
+        }
+        if (props.containsKey("hadoop.config.resources") || props.containsKey("hadoop.security.authentication")) {
+            return true;
+        }
+        return false;
+    }
+
+    private void loadFinalHdfsConfig(Map<String, String> origProps) {
+        if (MapUtils.isEmpty(origProps)) {
+            return;
+        }
+        finalHdfsConfig = new HashMap<>();
+        origProps.forEach((key, value) -> {
+            if (key.startsWith("hadoop.") || key.startsWith("dfs.") || key.equals("fs.defaultFS")) {
+                finalHdfsConfig.put(key, value);
+            }
+        });
+
+    }
+
+    @Override
+    protected String getResourceConfigPropName() {
+        return "hadoop.config.resources";
+    }
+
+    protected void checkRequiredProperties() {
+        super.checkRequiredProperties();
+        checkConfigFileIsValid(hadoopConfigResources);
+        if ("kerberos".equalsIgnoreCase(hdfsAuthenticationType)) {
+            if (Strings.isNullOrEmpty(hdfsKerberosPrincipal) || Strings.isNullOrEmpty(hdfsKerberosKeytab)) {
+                throw new IllegalArgumentException("HDFS authentication type is kerberos, "
+                        + "but principal or keytab is not set.");
+            }
+        }
+        if (StringUtils.isBlank(fsDefaultFS)) {
+            this.fsDefaultFS = HdfsPropertiesUtils.constructDefaultFsFromUri(origProps);
+        }
+    }
+
+    private void checkConfigFileIsValid(String configFile) {
+        if (Strings.isNullOrEmpty(configFile)) {
+            return;
+        }
+        loadConfigFromFile(getResourceConfigPropName());
+    }
+
+    public void toHadoopConfiguration(Configuration conf) {
```

Comment:
1. what is diff between `toHadoopConfiguration` and `getHadoopConfiguration`?
2. think twice about these methods. because we'd better load config only once instead of loading them every time